### PR TITLE
task 3 complete

### PR DESCRIPTION
**Pull request by Lordwhale**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -24,3 +24,21 @@ describe("GET /api", () => {
...
+        expect(topics).toHaveLength(3);
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            slug: expect.any(String),
+            description: expect.any(String),
+        });
+      });
+  });
+  })
+})
```
Comment (NCLukeHarrison) [inline __tests__/app.test.js:44]: Great test! I like that you've included a test for the length.

File: db/app.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const express = require("express");
+const app = express();
+const { getApi, getTopics } = require("./controllers/api.controller");
+
+app.use(express.json())
```
Comment (NCLukeHarrison) [inline db/app.js:5]: Not a good idea to include this before you need it.

File: db/controllers/api.controller.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+exports.getApi = (req, res) => {
+  res.status(200).send({ endpoints: endpointsJson });
+};
+
+exports.getTopics = (req, res) => {
+    selectTopics().then((rows) => {
+      res.status(200).send({ topics: rows });
+  })
+};
```
Comment (NCLukeHarrison) [inline db/controllers/api.controller.js:13]: Nice!

File: db/errors/errors.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    if (err.status && err.msg) {
+        res.status(err.status).send({ msg: err.msg });
+    } else {
+        next(err);
+    }
+};
+
+exports.serverErrorHandler = (err, req, res, next) => {
+    res.status(500).send({ msg: 'Internal Server Error' });
+};
```
Comment (NCLukeHarrison) [inline db/errors/errors.js:19]: I like that you've separated these errors out into a separate file. Nice!

File: db/models/model.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const db = require("../connection");
+
+exports.selectTopics = (res, req) => {
+    return db.query('SELECT * FROM topics;').then(({rows}) => {
+        return rows;
+    })
+};
```
Comment (NCLukeHarrison) [inline db/models/model.js:7]: Good stuff!

**Review by NCLukeHarrison — COMMENTED** [review]
Some great work here. It may be worth splitting your controllers and models up for each set of data, for example "topics.controllers.js" and then create a separate one for another data set. Don't forget to add a completed endpoint to your JSON file.

Nice work!

File: db/app.js (hunk 2 above)
Comment (Lordwhale) [reply db/app.js:5]: Line 5 removed, thank you.

**Pull request merged**